### ISSUE TITLE
Crash: "Cannot read property 'stamp' of null" in scale inside group mark

### ISSUE THREAD

**Issue by Jerrythafast** [vega/vega#424]
Dear Vega devs,

Here I am again with a graph spec that uses an odd combination of features that does not work well together! This time round I was filtering my data using a `signal`'s value, and feeding the remaining data into a `group` mark with a `facet` transform. I defined a `scale` inside the `group`. When I change the signal value such that an entire facet is filtered out, and then change the signal back to its initial value, Vega crashes when it evaluates the `domain` of the `scale`.

A minimal working example:
```json
{
  "signals": [{"name": "amplitude_threshold", "init": 5}],
  "data": [
    {
      "name": "table",
      "values": [
        {"name": "m1", "amplitude": 14},
        {"name": "m1", "amplitude":  3},
        {"name": "m1", "amplitude":  3},
        {"name": "m2", "amplitude":  7},
        {"name": "m2", "amplitude":  9},
        {"name": "m2", "amplitude":  4}
      ],
      "transform": [{"type": "filter", "test": "datum.amplitude >= amplitude_threshold"}]
    }
  ],
  "marks": [
    {
      "type": "group",
      "from": {
        "data": "table",
        "transform": [{"type": "facet", "groupby": ["name"]}]
      },
      "scales": [
        {
          "name": "x",
          "range": "width",
          "domain": {"field": "amplitude"}
        }
      ]
    }
  ]
}
```

If you load this spec into the Vega Editor, and then run the following commands:
```js
ved.view.signal("amplitude_threshold", 10).update();
ved.view.signal("amplitude_threshold", 5).update();
```
Vega crashes downstream of the second `update()` call on [line 375 of Scale.js](https://github.com/vega/vega/blob/master/src/scene/Scale.js#L375) because `data` is null at that point. Note how the first command changes `amplitude_threshold` to `10`, causing the entire `m2` facet to disappear.

Kind regards,
Jerry

**Comment by jheer** [comment]
Thanks Jerry. You deserve some kind of award for all the intricate bugs you've unearthed!

**Comment by arvind** [comment]
Another great catch @Jerrythafast! Your analysis was absolutely spot on. When Facet operator removed and recreated a new facet, the new nodes were queued for evaluation _after_ existing nodes responsible for constructing the group mark and calculate its scale transforms. 

I've pushed a fix in f953a52, which relies on vega-dataflow v1.3.0 (vega/vega-dataflow@80560b1). When  Facet observes new tuples, it triggers a re-ranking of the dataflow graph to ensure that new facet datasources continue to be computed _before_ scene graph nodes.

**Comment by arvind** [comment]
Shoot, that solution has broken the `jobs` example, so may not be a viable strategy.

**Comment by arvind** [comment]
Alright, and the subsequent issue should be fixed with c05a052. The re-ranking introduced in f953a52 required that the portion of the dataflow graph responsible for constructing the scenegraph be instantiated _after_ datasource operators have been evaluated.

**Comment by Jerrythafast** [comment]
Hey @arvind and @jheer, thanks for the hard work - closing four of my issues in one day! It almost makes me feel like I should try harder to keep you guys off the street ;)
